### DISCUSSION TITLE
Is it possible to apply cnoidal wave theory in HydroDyn?

### DISCUSSION THREAD

**Discussion by TheKite4** [OpenFAST/openfast#1486] [Q&A]
Hi, I want to apply cnoidal wave theory in OpenFAST, but it looks like only linear Airy theory is available for internal wave generation. I wonder if it's possible to use WaveMod=5 or 6 to apply externally produced cnoidal waves? 

If yes, is there any tool recommended to make those corresponding wave files (WvKinFile)?

**Comment by jjonkman** [comment]
Dear @TheKite4,

Technically the HydroDyn module of OpenFAST considers first-order wave theory (Airy) or first plus second-order wave theory (Sharma and Dean).  But I agree that HydroDyn does not support cnoidal wave theory.  As you suggested, you could input cnoidal waves via `WaveMod` = 6, but you would need to fully specify the wave kinematics within the `WvKinFile` files.  I'm not familiar with what software can generate such data.

Best regards,

**Reply by TheKite4** [reply]
Thank you for your fast response jjonkman!

Could you tell me the difference between WaveMod=6 and =5? As far as I know, =5 means a time series wave elevation profile, while =6 seems to give more detailed information including the kinematics of wave particles. I found that in the manual it says "The wave-elevation time series specified is assumed to be of first order and long-crested, but is not checked for physical correctness. " Is that why =5 is not capable of modelling a cnoidal wave which is a higher order wave? 

Cheers.

**Reply by jjonkman** [reply]
Dear @TheKite4,

Yes, that is correct.  The user-specified wave elevation entered for `WaveMod` = 5 is assuming to be a first-order (Airy) wave.  But with `WaveMod` = 6, HydroDyn makes no assumptions because the wave kinematics are fully user-specified.

Best regards,

**Reply by TheKite4** [reply]
Dear jjonkman:

Is there any test-run provided in OpenFAST that includes a WaveMod = 6 setting? Or do you have a template of the wave kinematics file?
Additionally, for the Morison theory joint and member settings in HydroDyn I am a bit confused. In my case which is a small-scale semi-submersible in shallow water, should I just assign the joints and members at the 3 columns from the bottom to SWL line vertically to the wave direction?

Cheers.

**Reply by jjonkman** [reply]
Dear @TheKite4,

I'm not aware of any OpenFAST test cases that support `WaveMod` = 6.  There were some tests in the old HydroDyn driver of FAST v8 that supported `WaveMod` = 6, but it doesn't appear that these have been ported over to the OpenFAST r-test per OpenFAST issue #179.  You can find the old HydroDyn driver tests from FAST v8 here: https://www.nrel.gov/wind/nwtc/hydrodyn.html, but these will need to be upgraded to work with OpenFAST.

Regarding modeling a semisubmersible in HydroDyn, I suggest using the NREL 5-MW baseline wind turbine top the OC4-DeepCwind semisubmersible as a working example: https://github.com/OpenFAST/r-test/tree/main/glue-codes/openfast/5MW_OC4Semi_WSt_WavesWN.

Best regards,
